src/executor/values.rs

```diff
@@ -18,44 +18,41 @@ pub struct ValuesExecutor {
 impl ValuesExecutor {
     #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
     pub async fn execute(self) {
-        for chunk in self.values.chunks(PROCESSING_WINDOW_SIZE) {
-            type Type = DataTypeKind;
-            // Create array builders.
-            let column_types = &self.column_types;
-            let mut builders = column_types
-                .iter()
-                .map(|ty| ArrayBuilderImpl::with_capacity(chunk.len(), ty))
-                .collect_vec();
-            // Push value into the builder.
-            let dummy = DataChunk::single(0);
-            for row in chunk {
-                for (expr, column_type, builder) in izip!(row, column_types, &mut builders) {
-                    let value = expr.eval(&dummy)?;
-                    let size = match column_type.kind {
-                        Type::Varchar(size) => size,
-                        Type::Char(size) => size,
-                        _ => None,
+        type Type = DataTypeKind;
+        let mut builder = DataChunkBuilder::new(self.column_types.iter(), PROCESSING_WINDOW_SIZE);
+        let dummy = DataChunk::single(0);
+        for ref row in self.values {
+            let mut row_data: Vec<DataValue> = Vec::with_capacity(row.len());
```

Comment:
As a matter of taste, this might be _arguably_ better:
```rust
let row: Result<Vec<_>, _> = izip!(row, &self.column_types)
    .map(...)
    .collect();
if let Some(chunk) = builder.push_row(row?) {
    yield chunk;
}
```
because the `for` loop only maps the data, without other side effects.

Reply:
Fixed. Thank you!